### PR TITLE
`COT` function

### PR DESCRIPTION
**Pull request by sarahyurick**
Looking at Issue #621 ... the changes so far allow for basic COT functionality with code like:

```
from dask_sql import Context
from dask.distributed import Client

client = Client()
c = Context()

c.sql("""
    SELECT COT(42) AS "the answer"
""", return_futures=False)
```

Mostly just wanting to check that there aren't any obvious issues with how I'm implementing this? Maybe want `da.tan()` instead of numpy?

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by charlesbluca** [comment]
add to allowlist

**Comment by sarahyurick** [comment]
Also looks like there are already tests for the trig functions in `tests/integration/test_rex.py` and `tests/integration/test_postgres.py` that are currently being skipped over. What's the protocol for adding unit tests in this case?

**Review by jdye64 — APPROVED** [review]
LGTM. Thanks for the PR!

One thing to note, I started this nasty pattern of placing these types of functions here in `sql.rs` and they will be moved to out DaskDialect implementation once we complete that. For now they are fine here.

**Comment by sarahyurick** [comment]
Ok thanks @jdye64 , makes sense! I'll update to use `da.tan()` instead of `np.tan()` and fix the style issue. Should I make a specific test for `COT` somewhere or just leave as is, since some tests already exist but are currently being skipped over?

File: dask_planner/src/sql.rs [hunk 1]
```diff
@@ -108,6 +108,11 @@ impl ContextProvider for DaskSQLContext {
             let rtf: ReturnTypeFunction = Arc::new(|_| Ok(Arc::new(DataType::Int64)));
             return Some(Arc::new(ScalarUDF::new("year", &sig, &rtf, &fun)));
         }
+        if "cot".eq(name) {
```
Comment (ayushdg) [inline dask_planner/src/sql.rs:111]: Question mainly for @jdye64. Do you have a preference on adding some functions to the dialect in rust vs registering it as a udf on the python side by creating a DaskFunction?

File: dask_sql/physical/rex/core/call.py [hunk 2]
```diff
@@ -943,6 +951,7 @@ class RexCallPlugin(BaseRexPlugin):
         # Temporary UDF functions that need to be moved after this POC
         "datepart": DatePartOperation(),
         "year": YearOperation(),
+        "cot": CotOperation(),
```
Comment (ayushdg) [inline dask_sql/physical/rex/core/call.py:954]: The existing implementation already has cot: https://github.com/dask-contrib/dask-sql/blob/f55c27664e5a06e9405cb1372a19ca60e0c53eae/dask_sql/physical/rex/core/call.py#L914

We might be able to get by without this and just having the parser recognize this function

**Review by ayushdg — CHANGES_REQUESTED** [review]

File: dask_sql/physical/rex/core/call.py (hunk 2 above)
Comment (sarahyurick) [reply dask_sql/physical/rex/core/call.py:954]: You're right - it still works without the additions to `call.py`, so I'll remove them now

File: dask_planner/src/sql.rs (hunk 1 above)
Comment (jdye64) [reply dask_planner/src/sql.rs:111]: I would prefer to have certain functions, like `cot`, which are quite common and in most DB engines be in the Rust dialect. However, once the Dialect is completed you will be able to interact with it through Python and add new `udfs` of course.

**Review by charlesbluca — COMMENTED** [review]
Thanks for the work here @sarahyurick 😄 to answer your question around testing, would it make sense to unskip the `COT` testing functions and comment out the queries / column assignments testing the unimplemented functions?

File: tests/integration/test_rex.py [hunk 3]
```diff
@@ -334,55 +333,56 @@ def test_math_operations(c, df):
             , ACOS(b) AS "acos"
             , ASIN(b) AS "asin"
             , ATAN(b) AS "atan"
-            , ATAN2(a, b) AS "atan2"
-            , CBRT(b) AS "cbrt"
```
Comment (charlesbluca) [inline tests/integration/test_rex.py:338]: We should be able to comment these out using [SQL comments](https://www.w3schools.com/Sql/sql_comments.asp), so with something like:

```sql
            -- , ATAN2(a, b) AS "atan2"
            -- , CBRT(b) AS "cbrt"
```

Comment (sarahyurick) [reply tests/integration/test_rex.py:338]: Good to know, thanks! I'll update it right now.

**Review by charlesbluca — APPROVED** [review]
LGTM! Looks like the failures here are related to CodeCov upload failures

**Review by ayushdg — APPROVED** [review]
lgtm!

**Pull request merged**